pytest/tests/sanity/block_sync_archival.py

```diff
@@ -97,11 +94,63 @@ def __exit__(self, *_):
                 node.cleanup()
 
 
-def get_all_blocks(node: cluster.BaseNode, *,
-                   head: cluster.BlockId) -> typing.Sequence[cluster.BlockId]:
+# TODO(#6458): Move this to separate file and merge with metrics module.
+def get_metrics(node_name: str,
+                node: cluster.BootNode) -> typing.Dict[str, int]:
+    """Fetches partial encoded chunk request count metrics from node.
+
+    Args:
+        node_name: Node’s name used when logging the counters.  This is purely
+            for debugging.
+        node: Node to fetch metrics from.
+
+    Returns:
+        A `{key: count}` dictionary where key is in ‘method/success’ format.
+        The values correspond to the
+        near_partial_encoded_chunk_request_processing_time_count Prometheus
+        metric.
+    """
+    url = 'http://{}:{}/metrics'.format(*node.rpc_addr())
+    response = requests.get(url)
+    response.raise_for_status()
+
+    metric_name = 'near_partial_encoded_chunk_request_processing_time'
```

Comment:
I'm surprised that this is not an argument of the function (but I guess you'll change it once you refactor?)

I'd also generalize this method a lot more (to simply return a dict of all metrics)

Reply:
Yes, that would be the idea. In this function it’s not worth generalising on the metric name since the function cannot even handle any other metric. Once this is refactored (as per #6458), we’d need a more general code which can handle any kind of metrics.